### DISCUSSION TITLE
Is there any way to get angular rotation relative to an object?

### DISCUSSION THREAD

**Discussion by Xeraster** [DanielChappuis/reactphysics3d#394] [General]
I'm trying to make a space ship game. My flight assist needs to be able to get the ship's angular velocity for pitch, yaw and roll. There is a nice little function called getAngularVelocity() for this. The only problem is this value is unsuitable for what i'm trying to do as it only gives absolute angular velocity and not relative to the ship's orientation.

I really need the pitch velocity to be in one axis, the roll velocity to be in another separate axis and the yaw velocity to be in a third separate axis, relative to the "front" (i guess) of the ship.

I've been trying to find some way, any way to convert this to relative/local angular velocities. I've messed with euler angle conversion/rotation functions, I've messed with both reactphysics3d quaternions and irrlicht quaternions. There just doesn't seem to be a way to convert absolute angular velocity to the angular velocity relative to the "cockpit" or "cockpit chair" of the ship.

I've been studying quaternions for the last few days trying to make sense of it but i'm getting nowhere. Does anyone know how I can get this working?

**Comment by DanielChappuis** [comment]
Hello,

I have converted your issue into a discussion here (please do not open issue if it's only a question). This is a math question and it is not directly related to the library. I let it open here is someone else wants to help but I prefer to spend time on questions related to the library itself. Maybe you could try to use ChatGPT or similar to help your convert the angular velocity into the local-frame of your player.
